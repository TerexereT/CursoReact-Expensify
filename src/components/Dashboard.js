import React from 'react'
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'

const ExpenseDBPage = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
)

export default ExpenseDBPage