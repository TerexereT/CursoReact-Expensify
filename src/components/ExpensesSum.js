import React from 'react'
import numeral from 'numeral'
import {connect} from 'react-redux'
import {Link} from 'react-router-dom'
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total'

export const ExpensesSummary = ({expenseCount, expensesTotal}) => {
    const aux = expenseCount === 1 ? 'expensa' : 'expensas'
    const formattedExpensesTotal = numeral(expensesTotal).format('$0,0.00')
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Viendo <span>{expenseCount}</span> {aux} de un total de <span>{formattedExpensesTotal}</span></h1>
                <div className="page-header__actions">
                    <Link className="button-layout" to="/create">Agregar expensa</Link>
                </div>
            </div>
        </div>
    )
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters)
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    }
}

export default connect (mapStateToProps)(ExpensesSummary)
