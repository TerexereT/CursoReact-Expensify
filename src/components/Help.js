import React from 'react'

const HelpExpenseDBPage = () => (
    <div>
        esto viene de Help Dashboard
    </div>
)

export default HelpExpenseDBPage