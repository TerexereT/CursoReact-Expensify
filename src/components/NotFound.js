import React from 'react'
import {Link} from 'react-router-dom'

const NotFoundDBPage = () => (
    <div>
        error 404! Page Not Found
        - <Link to="/">Home</Link>
    </div>
)

export default NotFoundDBPage