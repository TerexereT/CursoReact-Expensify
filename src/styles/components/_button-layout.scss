.button-layout {
    background: $blue;
    border: none;
    color: white;
    display: inline-block;
    font-size: $font-size-l;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}


.button--link {
    background: none;
    
}

.button-layout--secondary {
    @extend .button-layout;
    background: rgb(201, 0, 0);
}