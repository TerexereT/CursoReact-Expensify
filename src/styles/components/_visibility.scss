.show-for-mobile {
    @media (min-width: $desktopBp) {
        display: none;
    }
}

.show-for-desktop {
    @media (max-width: $desktopBp - .01rem){
        display: none;
    }
}