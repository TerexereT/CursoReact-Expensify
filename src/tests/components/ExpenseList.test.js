import React from 'react'
import {shallow} from 'enzyme'
import {ExpenseList} from '../../components/ExpenseList'
import expenses from '../fixtures/expenses'

test('render expenselist con expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses}/>)
    expect(wrapper).toMatchSnapshot()
})

test('render expenselist sin expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={[]}/>)
    expect(wrapper).toMatchSnapshot()
})

