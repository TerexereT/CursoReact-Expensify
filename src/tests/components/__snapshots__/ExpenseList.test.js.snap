// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render expenselist con expenses 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpenseList
    expenses={
        Array [
            Object {
              "amount": 2321,
              "createdAt": 0,
              "description": "hola",
              "id": "0",
              "note": "",
            },
            Object {
              "amount": 11321,
              "createdAt": -345600000,
              "description": "como",
              "id": "1",
              "note": "",
            },
            Object {
              "amount": 3321,
              "createdAt": 345600000,
              "description": "estas",
              "id": "2",
              "note": "",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ExpenseListItem
          amount={2321}
          createdAt={0}
          description="hola"
          id="0"
          note=""
/>,
        <ExpenseListItem
          amount={11321}
          createdAt={-345600000}
          description="como"
          id="1"
          note=""
/>,
        <ExpenseListItem
          amount={3321}
          createdAt={345600000}
          description="estas"
          id="2"
          note=""
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "amount": 2321,
          "createdAt": 0,
          "description": "hola",
          "id": "0",
          "note": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "amount": 11321,
          "createdAt": -345600000,
          "description": "como",
          "id": "1",
          "note": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "amount": 3321,
          "createdAt": 345600000,
          "description": "estas",
          "id": "2",
          "note": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ExpenseListItem
            amount={2321}
            createdAt={0}
            description="hola"
            id="0"
            note=""
/>,
          <ExpenseListItem
            amount={11321}
            createdAt={-345600000}
            description="como"
            id="1"
            note=""
/>,
          <ExpenseListItem
            amount={3321}
            createdAt={345600000}
            description="estas"
            id="2"
            note=""
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "amount": 2321,
            "createdAt": 0,
            "description": "hola",
            "id": "0",
            "note": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "amount": 11321,
            "createdAt": -345600000,
            "description": "como",
            "id": "1",
            "note": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "amount": 3321,
            "createdAt": 345600000,
            "description": "estas",
            "id": "2",
            "note": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`render expenselist sin expenses 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpenseList
    expenses={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        No Expenses
</p>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "No Expenses",
      },
      "ref": null,
      "rendered": "No Expenses",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          No Expenses
</p>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No Expenses",
        },
        "ref": null,
        "rendered": "No Expenses",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
